optuna/study/study.py

```diff
@@ -50,6 +50,7 @@
 
 
 ObjectiveFuncType = Callable[[trial_module.Trial], Union[float, Sequence[float]]]
+METRIC_NAMES = "study:metric_names"
```

Comment:
How about making this variable private to users and giving it a more descriptive name?

```suggestion
_SYSTEM_ATTR_METRIC_NAMES = "study:metric_names"
```